### PR TITLE
Add Github Action Super Linter 

### PR DESCRIPTION
**Pull request by WSLUser**
## Summary of the Pull Request

This uses the templates from https://github.com/github/super-linter/tree/master/TEMPLATES currently. A future PR can add the necessary templates to the Windows Terminal repository and update the source of Templates following the README. Additionally we can add flags to explicitly choose the linters applicable to this code base but is not necessary.

 
## References

https://github.com/microsoft/terminal/issues/7513

## PR Checklist
* [x] Closes https://github.com/microsoft/terminal/issues/7513
* [x] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/Terminal) and sign the CLA
* [ ] Tests added/passed N/A
* [ ] Documentation updated. If checked, please file a pull request on [our docs repo] (https://github.com/MicrosoftDocs/terminal) and link it here: N/A
* [ ] Schema updated. N/A
* [x] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: https://github.com/microsoft/terminal/issues/7513

## Detailed Description of the Pull Request / Additional comments

Per the README, this does not enforce any linting rules but rather outputs the suggestions in the build step, which are to be read by the PR submitter and Windows Terminal team to determine if they want to use the linting rule. C++ is currently not supported (Powershell, Json, Yaml, and Markdown will be the only things the linter checks for currently) but we could add our own custom support if desired in separate PR.

## Validation Steps Performed
It successfully runs. Currently only shows the yaml file itself being linted in this PR as a test case. It will apply to new PRs once this is merged. We can lint existing code base but would require a separate PR and examining the code output (also requires updating the yaml file temporarily).

### CODE REVIEW

**Comment by DHowett** [comment]
This is cool! Looking at the log in the workflow, it seems to only have picked up the yml file. Is it supposed to detect files that are new in this PR, or will it lint everything?

**Comment by WSLUser** [comment]
Just new PRs for now. It does have some customization options we can use. This PR mostly focuses on getting the basic support in. It only shows the yml file in this particular PR because this is the first PR with it being used in this repo.

**Comment by WSLUser** [comment]
Do we plan adding this in? Seems like there's plenty of doc PRs going around that could trivially be checked with this.

**Comment by miniksa** [comment]
I can't find an example of what this does. Dustin apparently saw something, but I rooted around in the logs and don't see it reporting any particular issues. Can you screencap it?

On principal, I'm okay with this because I like automatic linting conceptually.

**Comment by WSLUser** [comment]
Unhide checks and the very first one is the new listing build step called Lint Code Base. Their docs say use this for automated sync to their repo

File: .github/workflows/super-linter.report [hunk 1]
```diff
@@ -0,0 +1 @@
+
```
Comment (zadjii-msft) [inline .github/workflows/super-linter.report:1]: Is this file necessary for some reason?

Comment (WSLUser) [reply .github/workflows/super-linter.report:1]: It reports an error about the file missing. You can see it in the build steps. So I created it to get written to the next time it runs.

Comment (zadjii-msft) [reply .github/workflows/super-linter.report:1]: Wait does it though? Or am I looking at the wrong thing here?

```
2020-09-15 15:47:52 [INFO  ]   Generated reports:
2020-09-15 15:47:52 [INFO  ]   ----------------------------------------------
2020-09-15 15:47:52 [INFO  ]   ----------------------------------------------
2020-09-15 15:47:52 [INFO  ]   Reports generated in folder /github/workspace/super-linter.report
2020-09-15 15:47:52 [WARN  ]   Report output folder (/github/workspace/super-linter.report) does NOT exist.
```

That sounds like it wants a directory for that path, not a file. Also, it looks like the path should be `github/workspace/super-linter.report`, not `github/workflows/super-linter.report`.


https://github.com/github/super-linter seems to cocnur that it's supposed to be a path to a folder:

> ENV VAR | Default Value | Notes
> -- | -- | --
> OUTPUT_FOLDER | super-linter.report | The location where the output reporting will be generated to. Output folder must not previously exist.


But maybe I'm not understanding how that part works

Comment (WSLUser) [reply .github/workflows/super-linter.report:1]: Ah ok. I'll remove the file. I guess I didn't understand that part correctly.

**Review by zadjii-msft — APPROVED** [review]

**Review by DHowett — APPROVED** [review]

**Comment by DHowett** [comment]
Thanks for the contribution!

**Comment by DHowett** [comment]
Since this one was filed on Sept. 15, I don't think it's eligible for Hacktoberfest.
